aya/src/programs/mod.rs

```diff
@@ -136,6 +137,9 @@ pub enum ProgramError {
     #[error("unexpected program type")]
     UnexpectedProgramType,
 
+    #[error("invalid pin path `{error}`")]
+    InvalidPin { error: String },
```

Comment:
This is called `InvalidPin` and the map one is called `InvalidPath`. I think we should use the same name, perhaps `InvalidPinPath`?